broker/src/main/java/io/zeebe/broker/system/partitions/ZeebePartition.java

```diff
@@ -261,6 +274,19 @@ public void onFailure() {
         });
   }
 
+  @Override
+  public void onUnrecoverableFailure() {
+    actor.run(
+        () -> {
+          transitionToInactive();
```

Comment:
Is there a followup issue for updating the metrics - distinguish unhealthy vs dead?

Reply:
Yes, it's part of the "improve observability" issue which will make the dead status possible to see from the outside (incl metrics, topology, etc)